ImageInLib/ImageInLib/src/imageInterpolation.c

```diff
@@ -117,4 +119,29 @@ bool upSampling(dataType** originalImage, dataType** newImage, size_t length, si
     }
 
     return true;
-}
\ No newline at end of file
+}
+
+/*
+ * Get real world cordinate from image coordinate
+ * srcPoint : contains the voxel indexes
+ * realOrigin : image origin in real world
+ * imageSpacing : - distance between voxels in x and y direction
+ *                - distance between slices
+ * orientation : IJK(1,1,1), RAS(-1,-1,1) or LPS(1,-1,-1)
+*/
+Point3D imageCoordToRealCoord(imgPoint3D srcPoint, Point3D realOrigin, Spacing3D imageSpacing, Point3D orientation) { 
+    Point3D resultPoint;
+    resultPoint.x = realOrigin.x + orientation.x * imageSpacing.sx * srcPoint.x;
```

Comment:
we should have orientation consisting of three vectors (single for each direction) to have general implementation (considering skew slices)

Reply:
Yes I updates the function and the orientation. Also the new function for interpolation is ready/working with a really different approach in the implementation. With the new implementation the interpolation method can be a parameter and also finding the neighbors and the closest one will be in separate function. The final implementation is not done yet, but we can discuss the current code (which is working)